File: internal/dto/cronjob.go
package dto

type CronJobRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type CronJob struct {
	Name         string
	Namespace    string
	Schedule     string
	Timezone     string
	Suspend      bool
	Active       int
	LastSchedule string
	Age          string

	CreatedAt                  string
	Annotations                map[string]string
	ManagedFields              []ManagedField
	ConcurrencyPolicy          string
	SuccessfulJobsHistoryLimit int
	FailedJobsHistoryLimit     int
	LastSuccessfulTime         string
	LastSuccessfulTimeAt       string
	LastScheduleAt             string
	JobParallelism             int
	JobCompletions             string
	JobSuspend                 bool
	JobTTLSecondsAfterFinished int32
}

File: packages/core/frontend/src/types/resources/cronjob.ts
import type { ManagedField } from "./shared";

export interface CronJob {
  Name: string;
  Namespace: string;
  Schedule: string;
  Timezone: string;
  Suspend: boolean;
  Active: number;
  LastSchedule: string;
  Age: string;
  CreatedAt: string;
  Annotations?: Record<string, string>;
  ManagedFields: ManagedField[];
  ConcurrencyPolicy: string;
  SuccessfulJobsHistoryLimit: number;
  FailedJobsHistoryLimit: number;
  LastSuccessfulTime?: string;
  LastSuccessfulTimeAt?: string;
  LastScheduleAt?: string;
  JobParallelism: number;
  JobCompletions: string;
  JobSuspend: boolean;
  JobTTLSecondsAfterFinished: number;
}

export interface CronJobSummary {
  Scheduled: number;
  Suspended: number;
}
